server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSWriteOptionsCheckUtility.java

```diff
@@ -0,0 +1,145 @@
+              put(CICSLexer.PFXLENG, ErrorSeverity.ERROR);
+              put(CICSLexer.WAIT, ErrorSeverity.WARNING);
+              put(CICSLexer.OPERATOR, ErrorSeverity.ERROR);
+              put(CICSLexer.TEXT, ErrorSeverity.ERROR);
+              put(CICSLexer.TEXTLENGTH, ErrorSeverity.ERROR);
+              put(CICSLexer.ROUTECODES, ErrorSeverity.ERROR);
+              put(CICSLexer.NUMROUTES, ErrorSeverity.ERROR);
+              put(CICSLexer.CONSNAME, ErrorSeverity.ERROR);
+              put(CICSLexer.EVENTUAL, ErrorSeverity.WARNING);
+              put(CICSLexer.ACTION, ErrorSeverity.ERROR);
+              put(CICSLexer.CRITICAL, ErrorSeverity.WARNING);
+              put(CICSLexer.IMMEDIATE, ErrorSeverity.WARNING);
+              put(CICSLexer.REPLY, ErrorSeverity.ERROR);
+              put(CICSLexer.MAXLENGTH, ErrorSeverity.ERROR);
+              put(CICSLexer.REPLYLENGTH, ErrorSeverity.ERROR);
+              put(CICSLexer.TIMEOUT, ErrorSeverity.ERROR);
+            }
+          };
+
+  public CICSWriteOptionsCheckUtility(DialectProcessingContext context, List<SyntaxError> errors) {
+    super(context, errors, DUPLICATE_CHECK_OPTIONS);
+  }
+
+  /**
+   * Entrypoint to check CICS Write rule options
+   *
+   * @param ctx ParserRuleContext subclass containing options
+   * @param <E> A subclass of ParserRuleContext
+   */
+  public <E extends ParserRuleContext> void checkOptions(E ctx) {
+    switch (ctx.getRuleIndex()) {
+      case CICSParser.RULE_cics_write_file:
+        checkWriteFile((CICSParser.Cics_write_fileContext) ctx);
+        break;
+      case CICSParser.RULE_cics_write_journalname:
+        checkWriteJournalname((CICSParser.Cics_write_journalnameContext) ctx);
+        break;
+      case CICSParser.RULE_cics_write_operator:
+        checkWriteOperator((CICSParser.Cics_write_operatorContext) ctx);
+        break;
+      default:
+        break;
+    }
+    checkDuplicates(ctx);
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkWriteFile(CICSParser.Cics_write_fileContext ctx) {
+    checkHasExactlyOneOption("FILE or DATASET", ctx, ctx.FILE(), ctx.DATASET());
+    checkHasMandatoryOptions(ctx.FROM(), ctx, "FROM");
+    checkHasMandatoryOptions(ctx.RIDFLD(), ctx, "RIDFLD");
+    if (!ctx.SYSID().isEmpty()) {
+      checkHasMandatoryOptions(ctx.LENGTH(), ctx, "LENGTH");
+    }
+    checkHasMutuallyExclusiveOptions("RBA or RRN or XRBA", ctx.RBA(), ctx.RRN(), ctx.XRBA());
+  }
+
+  private void checkWriteJournalname(CICSParser.Cics_write_journalnameContext ctx) {
+    checkHasMandatoryOptions(ctx.JOURNALNAME(), ctx, "JOURNALNAME");
+    checkHasMandatoryOptions(ctx.JTYPEID(), ctx, "JTYPEID");
+    checkHasMandatoryOptions(ctx.FROM(), ctx, "FROM");
+    if (!ctx.PFXLENG().isEmpty()) {
+      checkHasMandatoryOptions(ctx.PREFIX(), ctx, "PREFIX");
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkWriteOperator(CICSParser.Cics_write_operatorContext ctx) {
```

Comment:
Looks Operator is mandatory here
![image](https://github.com/user-attachments/assets/9baa6618-d5cd-4900-a0b3-36584de9b3d1)

or it can be mutual exclusive among FILE OR JOURNALNAME OR JOURNALNUM OR OPERATOR